src/web/middleware/jira/jira-query-symmetric-jwt-auth-middleware.ts

```diff
@@ -0,0 +1,38 @@
+import type { NextFunction, Request, RequestHandler, Response } from 'express';
+
+import { jiraContextSymmetricJwtTokenVerifier } from '../../../infrastructure/jira/inbound-auth';
+import { UnauthorizedResponseStatusError } from '../../errors';
+
+/**
+ * Authenticates requests using a symmetric JWT token received in a query param.
+ *
+ * In case of successful authentication, `connectInstallation` and `atlassianUserId` are set in locals.
+ *
+ * @remarks
+ * Context JWT tokens are sent in a query param when loading an iframe
+ *
+ * @see https://developer.atlassian.com/cloud/jira/platform/understanding-jwt-for-connect-apps/#types-of-jwt-token
+ * @see https://community.developer.atlassian.com/t/action-required-atlassian-connect-vulnerability-allows-bypass-of-app-qsh-verification-via-context-jwts/47072
+ */
+export const jiraQuerySymmetricJwtAuthMiddleware: RequestHandler = (
```

Comment:
**nitpick (non-blocking):** Consider renaming this to `jiraContextSymmetricJwtFromQueryAuthMiddleware` to highlight that it verifies a [context symmetric JWT](https://developer.atlassian.com/cloud/jira/platform/understanding-jwt-for-connect-apps/#types-of-jwt-token).

We could also name our middlewares based on their purpose  instead of the token type (e.g., `staticPageAuthMiddleware`) but we will need to rename other middlewares as well for consistency.